gam/clustering.py

```diff
@@ -549,16 +576,32 @@ def _finalize(self, count, mean, m2):
         Returns:
             variance (int): The variance of the medoids
         """
-        mean = mean
-        variance = (m2 / count)
+        variance = m2 / count
         sample_variance = m2 / (count - 1)
         if count < 2:
             return float("nan")
         else:
-            return mean, variance, sample_variance
+            return variance, sample_variance
 
-    def _bandit_search_singles(self, X, dist_func, d_nearest, tmp_arr, j, i):
-        """Inner loop for bandit build functions.
+    def _dask_finalize(self, count, m2):
+        """Finding variance for each new mean
+
+        Args:
+            count (int): The number of reference points.
+            m2 (int): The updated mean.
+
+        Returns:
+            variance (int): The variance of the medoids
+        """
+        variance = da.divide(m2, count)
+        sample_variance = da.divide(m2, (count - 1))
+        if count < 2:
+            return float("nan")
+        else:
+            return variance, sample_variance
```

Comment:
if `count` and `m2` are ints, I'm not sure it makes sense to use dask operations here.